cmd/libs/go2idl/client-gen/generators/generator_for_type.go

```diff
@@ -73,13 +74,25 @@ func (g *genClientForType) GenerateType(c *generator.Context, t *types.Type, w i
 		"Package":           namer.IC(pkg),
 		"Group":             namer.IC(g.group),
 		"watchInterface":    c.Universe.Type(types.Name{Package: "k8s.io/kubernetes/pkg/watch", Name: "Interface"}),
-		"apiDeleteOptions":  c.Universe.Type(types.Name{Package: "k8s.io/kubernetes/pkg/api", Name: "DeleteOptions"}),
-		"apiListOptions":    c.Universe.Type(types.Name{Package: "k8s.io/kubernetes/pkg/api", Name: "ListOptions"}),
 		"apiParameterCodec": c.Universe.Type(types.Name{Package: "k8s.io/kubernetes/pkg/api", Name: "ParameterCodec"}),
 		"PatchType":         c.Universe.Type(types.Name{Package: "k8s.io/kubernetes/pkg/api", Name: "PatchType"}),
 		"namespaced":        namespaced,
 	}
 
+	if g.version == "unversioned" {
+		m["DeleteOptions"] = c.Universe.Type(types.Name{Package: "k8s.io/kubernetes/pkg/api", Name: "DeleteOptions"})
+		m["ListOptions"] = c.Universe.Type(types.Name{Package: "k8s.io/kubernetes/pkg/api", Name: "ListOptions"})
+	} else {
```

Comment:
I would suggest you check for g.version == "v1" here and fail on any other value. This would remind us to fix this when we have v2 api.